h2o-hive/src/main/java/water/hive/DelegationTokenPrinter.java

```diff
@@ -0,0 +1,43 @@
+package water.hive;
+
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.security.token.delegation.DelegationTokenIdentifier;
+import org.apache.hadoop.security.token.Token;
+import org.apache.hadoop.security.token.delegation.AbstractDelegationTokenIdentifier;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Objects;
+
+public class DelegationTokenPrinter {
+
+    private static class PrintingDelegationTokenIdentifier extends AbstractDelegationTokenIdentifier {
+        public Text getKind() {
+            return new Text("PRINT");
+        }
+    }
+
+    public static void printToken(final String tokenString) {
+        Objects.requireNonNull(tokenString);
+        try {
+            final Token<DelegationTokenIdentifier> token = new Token<>();
+            token.decodeFromUrlString(tokenString);
+            final AbstractDelegationTokenIdentifier identifier = new PrintingDelegationTokenIdentifier();
```

Comment:
is there a reason you cannot just call `Token#decodeIdentifier`?

Reply:
yes, for some distros/versions this call fails because the token class is not available in the jdbc driver JAR, thats why I am using my own Identifier class to load the data